providers/dns/hurricanedns/hurricanedns_test.go

```diff
@@ -0,0 +1,60 @@
+package hurricanedns
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	heUsername string
+	hePassword string
+	heDomain   string
+	heLiveTest bool
+)
+
+func init() {
+	heUsername = os.Getenv("HE_USERNAME")
+	hePassword = os.Getenv("HE_PASSWORD")
+	heDomain = os.Getenv("HE_DOMAIN_NAME")
+
+	if len(heUsername) > 0 && len(hePassword) > 0 && len(heDomain) > 0 {
+		heLiveTest = true
+	}
+}
+
+func TestNewDNSProvider(t *testing.T) {
+	provider, err := NewDNSProvider()
+
+	if !heLiveTest {
+		assert.Error(t, err)
+	} else {
+		assert.NotNil(t, provider)
+		assert.NoError(t, err)
+	}
+}
```

Comment:
Could add a test as the following example:

https://github.com/xenolf/lego/blob/c4bbb4b819c087d02066fe745d7ca5b60a480d36/providers/dns/cloudflare/cloudflare_test.go#L51-L58